go/vt/tabletserver/schema_info.go

```diff
@@ -330,14 +321,7 @@ func (si *SchemaInfo) ClearQueryPlanCache() {
 }
 
 // CreateOrUpdateTable must be called if a DDL was applied to that table.
-func (si *SchemaInfo) CreateOrUpdateTable(ctx context.Context, tableName sqlparser.TableIdent) error {
-	si.actionMutex.Lock()
-	defer si.actionMutex.Unlock()
-	return si.createOrUpdateTableLocked(ctx, tableName.String())
-}
-
-// createOrUpdateTableLocked must only be called while holding actionMutex.
-func (si *SchemaInfo) createOrUpdateTableLocked(ctx context.Context, tableName string) error {
+func (si *SchemaInfo) CreateOrUpdateTable(ctx context.Context, tableName string) error {
```

Comment:
This function is public and doesn't require si.mu. It actually obtains it later below just before making the schema change.